tests/Entity/ProcurationProxyTest.php

```diff
@@ -10,22 +10,69 @@
 
 class ProcurationProxyTest extends TestCase
 {
+    /**
+     * @dataProvider provideTestCases
+     */
+    public function testGetAvailableRoundsCount(
+        int $proxiesCount,
+        int $requestElectionRoundsCount,
+        int $proxyElectionRoundsCount,
+        int $expectedAvailableElectionRoundsCount
+    ) {
+        $maxElectionRound = $requestElectionRoundsCount > $proxyElectionRoundsCount
+            ? $requestElectionRoundsCount
+            : $proxyElectionRoundsCount
+        ;
+
+        $electionRounds = [];
+        for ($i = 0; $i < $maxElectionRound; ++$i) {
+            array_push($electionRounds, new ElectionRound());
+        }
+
+        $request = new ProcurationRequest();
+        $request->setElectionRounds(\array_slice($electionRounds, 0, $requestElectionRoundsCount));
+
+        $proxy = new ProcurationProxy(null);
+        $proxy->setProxiesCount($proxiesCount);
+        $proxy->addFoundRequest($request);
+        $proxy->setElectionRounds(\array_slice($electionRounds, 0, $proxyElectionRoundsCount));
+
+        $this->assertCount($expectedAvailableElectionRoundsCount, $available = $proxy->getAvailableRounds());
+    }
+
+    public function provideTestCases(): \Generator
+    {
+        yield [1, 1, 1, 0];
```

Comment:
Can you just add comments by "block" to explain in few words ? 

Or maybe you can add a description by `yield` like : 

```php
yield 'Your beautiful description' => [1, 1, 2, 1];
```